microcloud/cmd/microcloud/main_init.go

```diff
@@ -532,6 +539,83 @@ func validateSystems(s *service.Handler, systems map[string]InitSystem) (err err
 	return nil
 }
 
+// checkClustered checks whether any of the selected systems have already initialized a service.
+// Returns the first system we find that is initialized for the given service, along with all of that system's existing cluster members.
+func checkClustered(s *service.Handler, autoSetup bool, serviceType types.ServiceType, systems map[string]InitSystem) (firstInitializedSystem string, existingMembers map[string]string, err error) {
+	// LXD should always be uninitialized at this point, so we can just return default values that consider LXD uninitialized.
+	if serviceType == types.LXD {
+		return "", nil, nil
+	}
+
+	for peer, system := range systems {
+		var remoteClusterMembers map[string]string
+		var err error
+
+		// If the peer in question is ourselves, we can just use the unix socket.
+		if peer == s.Name {
+			remoteClusterMembers, err = s.Services[serviceType].ClusterMembers(context.Background())
+		} else {
+			remoteClusterMembers, err = s.Services[serviceType].RemoteClusterMembers(context.Background(), system.ServerInfo.AuthSecret, system.ServerInfo.Address)
+		}
+
+		if err != nil && err.Error() != "Daemon not yet initialized" {
+			return "", nil, fmt.Errorf("Failed to reach %s on system %q: %w", serviceType, peer, err)
+		}
+
+		// If we failed to retrieve cluster members due to the system not being initialized, we can ignore it.
+		if err != nil {
+			continue
+		}
+
+		clusterMembers := map[string]string{}
+		for k, v := range remoteClusterMembers {
+			host, _, err := net.SplitHostPort(v)
+			if err != nil {
+				return "", nil, err
+			}
+
+			clusterMembers[k] = host
+		}
+
+		if autoSetup {
```

Comment:
As `autoSetup` doesn't get modified in this func, would it be a better choice not passing this var into `checkClustered` and performing the check outside?

Reply:
It's just for efficiency. We can move it outside but that would mean that we would have to run through the rest of `checkClustered`'s implementation. Where it is now, as soon as at least 1 system gets to this point, we can exit if `autoSetup` is true, rather than continuing to build the list and checking more systems, only to error out afterward anyway.